src/grid.js

```diff
@@ -319,6 +320,19 @@ Grid.prototype.setSymmetry = function(symmetry) {
   this.sanitize();
 }
 Grid.prototype.setSize = function(width, height) {
+  // If the start and end are in the normal place, erase them and re-add them
+  // in the new normal places.
```

Comment:
As an alternative logic, we could see if there is exactly one start element and, if so, clamp it to the grid. (Particularly this might happen once we randomize starting arrangement, which is threatened by TODOs elsewhere in Grid.)

But this should work very well for now.